pkgs/applications/virtualization/podman/default.nix

```diff
@@ -47,26 +47,31 @@ buildGoModule rec {
   buildPhase = ''
     runHook preBuild
     patchShebangs .
-    ${if stdenv.isDarwin
-      then "make podman-remote"
-      else "make podman"}
+    ${if stdenv.isDarwin then ''
+      make podman-remote # podman-mac-helper uses FHS paths
+    '' else ''
+      make bin/podman bin/rootlessport
+    ''}
     make docs
     runHook postBuild
   '';
 
   installPhase = ''
     runHook preInstall
-  '' + lib.optionalString stdenv.isDarwin ''
-    mv bin/{darwin/podman,podman}
-  '' + ''
-    install -Dm555 bin/podman $out/bin/podman
-    installShellCompletion --bash completions/bash/*
-    installShellCompletion --fish completions/fish/*
-    installShellCompletion --zsh completions/zsh/*
-    MANDIR=$man/share/man make install.man-nobuild
-    install -Dm644 cni/87-podman-bridge.conflist -t $out/etc/cni/net.d
-    install -Dm644 contrib/tmpfile/podman.conf -t $out/lib/tmpfiles.d
-    install -Dm644 contrib/systemd/system/podman.{socket,service} -t $out/lib/systemd/system
+    ${if stdenv.isDarwin then ''
+      mv bin/{darwin/podman,podman}
+    '' else ''
+      install -Dm644 cni/87-podman-bridge.conflist -t $out/etc/cni/net.d
+      install -Dm644 contrib/tmpfile/podman.conf -t $out/lib/tmpfiles.d
+      for s in contrib/systemd/**/*.in; do
+        substituteInPlace "$s" --replace "@@PODMAN@@" "podman" # don't use unwrapped binary
```

Comment:
Hi @zowoq :wave:

I would like to use the `podman` user units provided by nixpkgs but whenever I use `podman` through its socket I get the following error:

```
podman.service: Failed to locate executable podman: No such file or directory
```

If I modify the user unit to include the absolute path of `podman` everything works flawlessly. It seems like I might be able to tweak this code a bit to patch the right paths in:

https://github.com/NixOS/nixpkgs/blob/1ffba9f2f683063c2b14c9f4d12c55ad5f4ed887/pkgs/applications/virtualization/podman/wrapper.nix#L76

What do you think?

Reply:
Can't it be fixed in the module rather than in the wrapper?

Reply:
My scenario is Ubuntu.

Reply:
Would setting the paths in the wrapper work with `extraPackages` from the module?

Reply:
Unfortunately not, from the best of my ability to reason about the package, because it doesn't touch the systemd unit.

Reply:
https://github.com/NixOS/nixpkgs/pull/218355

@aanderse We've stopped using the wrapper so systemd units should work without changes on non-nixos.

Reply:
I really appreciate the ping! Thanks.